woodwork/deserialize.py

```diff
@@ -82,15 +82,19 @@ def _typing_information_to_woodwork_table(table_typing_info, validate, **kwargs)
         column_metadata[col_name] = col['metadata']
         use_standard_tags[col_name] = col['use_standard_tags']
 
-        if col['physical_type']['type'] == 'category':
+        col_type = col['physical_type']['type']
+        if col_type == 'category':
             # Make sure categories are recreated properly
             cat_values = col['physical_type']['cat_values']
             cat_dtype = col['physical_type']['cat_dtype']
             if table_type == 'pandas':
                 cat_object = pd.CategoricalDtype(pd.Index(cat_values, dtype=cat_dtype))
             else:
                 cat_object = pd.CategoricalDtype(pd.Series(cat_values))
-            category_dtypes[col_name] = cat_object
+            col_type = cat_object
+        elif table_type == 'koalas' and col_type == 'object':
+            col_type = 'string'
+        category_dtypes[col_name] = col_type
```

Comment:
Minor point, but this dictionary is no longer being used to store just category dtypes, so I think it would be more clear what is happening if we update the name to something like `column_dtypes`